lib/LedgerSMB/Scripts/import_csv.pm

```diff
@@ -428,32 +441,41 @@ our $process = {
     inventory_multi => \&_process_inventory_multi,
 };
 
-=head2 parse_file
+=head2 _parse_file
 
 This parses a file, and returns a the csv in tabular format.
 
 =cut
 
-sub parse_file {
+sub _parse_file {
     my $self = shift @_;
 
     my $handle = $self->{_request}->upload('import_file');
     my $contents = join("\n", <$handle>);
+    my $sep = "";
+    my $n = @{$cols->{$self->{type}}}-1; # 1 separator less than fields
 
     $self->{import_entries} = [];
     for my $line (split /(\r\n|\r|\n)/, $contents){
-        next if ($line !~ /,/);
+        if ( $sep eq "" ) {
+            if    ($n == (() = $line =~  /,/g)) { $sep =  "," }
+            elsif ($n == (() = $line =~  /;/g)) { $sep =  ";" }
+            elsif ($n == (() = $line =~ /\t/g)) { $sep = "\t" }
+            else  { die "Unrecognized file format" }
```

Comment:
When using heuristics like this, there are many occasions where we can go wrong. I'd prefer to "simply" use settings from upload dialog; after all, this is an interface type of activity and that's exactly where you want things to be deterministic.